### PR TITLE
TRIB-154: Improve backend standard compliance by documenting TribeApp backend

### PR DESCRIPTION
**Pull request by maryamkhan14**
Currently, the backend has no documentation. This makes it hard to remain consistent about endpoint naming and the responses generated. Additionally, the Postman collection has to be generated separately, so it easily gets outdated ([TRIB-141](url)). As part of improving backend standard compliance [per TRIB-154](https://mockprogrammingjob.atlassian.net/browse/TRIB-154), this PR addresses both those issues by implementing Swagger/OpenAPI documentation on the backend. 

SpringDoc is used to generate the Swagger UI and the JSON OpenAPI spec. Since SpringDoc requires verbose annotations that impact code readability, this PR configures custom, concise annotations that achieve the same effect as the original SpringDoc annotations. These custom annotations are also flexible: a developer may borrow parts of an original SpringDoc annotation, if needed. Of course, a developer may also use the entire original SpringDoc annotation, although it would be rarely necessary.

To prevent clutter in the controller classes, all annotations documenting an endpoint are compiled into a single annotation, stored in the annotations package. This single annotation is applied to the endpoint it relates to. The complete documentation on using the annotations is available [here](https://tribeapp.notion.site/c81772c35c164a98bfe0c970eb5666ce?v=70e795432cd04144b84b8a392bfecb2e&pvs=4).

In summary, this PR makes the following changes:
- Add dependencies needed for Springdoc and OpenAPI
- Specify security schemes and general information about backend API
- Customize Springdoc implementation to process concise annotation usage
- Create custom annotations for endpoint request bodies & responses
- Annotate controllers, DTOs, and entities (if returned by endpoint)

### CODE REVIEW

**Comment by haxwell** [comment]
Hi Maryam, thank you again for this incredible work. :pray: 

There are just a couple merge issues to be addressed. I handled some of the conflicts, but I believe you'll need to sync your fork, and merge those changes to get your build to complete.

**Review by haxwell — CHANGES_REQUESTED** [review]
Sync with the main fork, and get the build running.

**Comment by maryamkhan14** [comment]
> Sync with the main fork, and get the build running.

It should work now! Just needed to make 2 small changes to reconcile differences between the develop and swagger-documentation branches.

**Review by haxwell — APPROVED** [review]
lgtm
